src/lang/builtins_bool.ml

```diff
@@ -27,9 +27,29 @@ let () =
   let register_op name op =
     add_builtin name ~cat:Bool ~descr:"Comparison of comparable values."
       [("", t, None, None); ("", t, None, None)] Lang.bool_t (fun p ->
-        let a = Lang.assoc "" 1 p |> Lang.demeth in
-        let b = Lang.assoc "" 2 p |> Lang.demeth in
-        Lang.bool (op (Lang.compare_values a b)))
+        let a = Lang.assoc "" 1 p in
+        let b = Lang.assoc "" 2 p in
+        let a' = Lang.demeth a in
+        let b' = Lang.demeth b in
+        (* For records, we also compare fields. *)
```

Comment:
Why only for records?

Reply:
My idea is that we would tend to write something like
```
if http.get("http://...") == "bla" then
  ...
end
```
to compare the result of getting a page. And we don't wan't to take methods in accounts in this case.

And most of the time, excepting for records, fields are functions, which we cannot compare...